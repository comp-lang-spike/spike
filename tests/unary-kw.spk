#!/usr/bin/env spike

class Foo : Object {
    obj answer;
    
    self init answer: a {
        answer = a;
    }
    
    self zoo {
        return answer;
    }
    self mutate: a {
        return a + 2;
    }
    
} meta {
    self new {
        return self new answer: 42;
    }
    self new answer: a {
        return (super new) init answer: a;
    }
}

main(argv) {
    obj f, a;
    
    f = (Foo) new;
    a = f zoo;
    printf("zoo: %d\n", a);
    a = f mutate: (f zoo);
    printf("mutation: %d\n", a);
    
    f = (Foo) new answer: 40;
    a = f zoo;
    printf("zoo: %d\n", a);
    a = f mutate: (f zoo);
    printf("mutation: %d\n", a);
    
    return 0;
}
