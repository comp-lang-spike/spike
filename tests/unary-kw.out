zoo: 42
mutation: 44
zoo: 40
mutation: 42
